src/compiler/declarationEmitter.ts

```diff
@@ -1888,7 +1888,7 @@ namespace ts {
     /* @internal */
     export function writeDeclarationFile(declarationFilePath: string, sourceFileOrBundle: SourceFile | Bundle, host: EmitHost, resolver: EmitResolver, emitterDiagnostics: DiagnosticCollection, emitOnlyDtsFiles: boolean) {
         const emitDeclarationResult = emitDeclarations(host, resolver, emitterDiagnostics, declarationFilePath, sourceFileOrBundle, emitOnlyDtsFiles);
-        const emitSkipped = emitDeclarationResult.reportedDeclarationError || host.isEmitBlocked(declarationFilePath) || host.getCompilerOptions().noEmit;
+        const emitSkipped = emitDeclarationResult.reportedDeclarationError || host.isEmitBlocked(declarationFilePath);

```

Comment:
> @mhegazy Do we want to emit declaration files for `--declaration --noEmit` for TypeScript (i.e. without `--allowJs`)? This would be a breaking change and could have undesired consequences. Do we need to have a new compiler option for this behavior?

I agree, we should not do that. either we emit declarations for Js files with  `--declaration --allowjs`, and users have to pass a `--outDir tmp`, then ignore it (this is still potentially a breaking change since we did not emit these files in the past so current `--allowjs` users would find this surprising), or we add a new flag `--emitDeclarationsOnly`.